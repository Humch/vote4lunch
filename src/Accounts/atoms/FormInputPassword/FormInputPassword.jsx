import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import PropTypes from 'prop-types';

import { formInputPassword, formInputPasswordRepeater } from '../../actions';

class FormInputPassword extends Component {
  constructor(props) {
    super(props);

    this.onChangeValue = this.onChangeValue.bind(this);
  }

  onChangeValue(value) {
    const {
      repeater,
      formInputPassword: formInputPasswordAction,
      formInputPasswordRepeater: formInputPasswordRepeaterAction,
    } = this.props;

    if (!repeater) {
      formInputPasswordAction(value);
    } else {
      formInputPasswordRepeaterAction(value);
    }
  }

  render() {
    const {
      repeater,
      password,
      passwordRepeater,
    } = this.props;
    let renderExist = '';

    if (
      password !== passwordRepeater
      && password !== ''
      && passwordRepeater !== ''
    ) {
      renderExist = 'is-invalid';
    }

    return (
      <input
        type="password"
        name={!repeater ? 'password' : 'passwordRepeater'}
        id={!repeater ? 'password' : 'passwordRepeater'}
        value={!repeater ? password : passwordRepeater}
        onChange={e => this.onChangeValue(e.target.value)}
        required
        className={`form-control ${renderExist}`}
      />
    );
  }
}

FormInputPassword.propTypes = {
  repeater: PropTypes.bool,
  password: PropTypes.string.isRequired,
  passwordRepeater: PropTypes.string.isRequired,
  formInputPassword: PropTypes.func.isRequired,
  formInputPasswordRepeater: PropTypes.func.isRequired,
};

const mstp = ({ userData }) => ({
  password: userData.password,
  passwordRepeater: userData.passwordRepeater,
});

const mdtp = dispatch => bindActionCreators({
  formInputPassword,
  formInputPasswordRepeater,
}, dispatch);

export default connect(mstp, mdtp)(FormInputPassword);
