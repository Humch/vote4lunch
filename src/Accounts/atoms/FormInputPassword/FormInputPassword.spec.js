/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import FormInputPassword from './FormInputPassword';

describe('FormInputPassword Snapshot', () => {
  const initialState = {
    userData: {
      password: '',
      passwordRepeater: '',
    },
    passwordChecker: {
      tooShort: false,
    },
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of FormInputPassword with repeater prop false', () => {
    const renderedValue = renderer.create(<FormInputPassword store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  it('capturing Snapshot of FormInputPassword with repeater prop true', () => {
    const renderedValue = renderer.create(<FormInputPassword repeater store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});


describe('FormInputPassword (Shallow + passing the {store} directly)', () => {
  const initialState = {
    userData: {
      password: '',
      passwordRepeater: '',
    },
    passwordChecker: {
      tooShort: false,
    },
  };
  const mockStore = configureStore();
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<FormInputPassword store={store} />);
  });

  it('render the connected(SMART) component', () => {
    expect(container.length).toEqual(1);
  });

  it('check Prop matches with initialState', () => {
    expect(container.prop('password')).toEqual(initialState.userData.password);
    expect(container.prop('passwordRepeater')).toEqual(initialState.userData.passwordRepeater);
  });
});

describe('FormInputEmail - REACT-REDUX (Mount + wrapping in <Provider>) without repeater prop', () => {
  const initialState = {
    userData: {
      password: '',
      passwordRepeater: '',
    },
    passwordChecker: {
      tooShort: false,
    },
  };
  const mockStore = configureStore();
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    store.clearActions();
    wrapper = mount(<Provider store={store}><FormInputPassword /></Provider>);
  });

  it('render the connected(SMART) component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('check onChange in password input', () => {
    const input = wrapper.find('#password');
    expect(input.props().value).toBe(initialState.userData.password);

    const event = { target: { value: 'f4k3pAssw0rd' } };
    input.props().onChange(event);
    const action = store.getActions();
    expect(action[0].type).toBe('FORM_INPUT_PASSWORD');
    expect(action[0].password).toBe('f4k3pAssw0rd');
  });
});

describe('FormInputEmail - REACT-REDUX (Mount + wrapping in <Provider>) with repeater prop', () => {
  const initialState = {
    userData: {
      password: '',
      passwordRepeater: '',
    },
    passwordChecker: {
      tooShort: false,
    },
  };
  const mockStore = configureStore();
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    store.clearActions();
    wrapper = mount(<Provider store={store}><FormInputPassword repeater /></Provider>);
  });

  it('render the connected(SMART) component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('check onChange in password input', () => {
    const input = wrapper.find('#passwordRepeater');
    expect(input.props().value).toBe(initialState.userData.passwordRepeater);

    const event = { target: { value: 'f4k3pAssw0rd' } };
    input.props().onChange(event);
    const action = store.getActions();
    expect(action[0].type).toBe('FORM_INPUT_PASSWORD_REPEATER');
    expect(action[0].passwordRepeater).toBe('f4k3pAssw0rd');
  });
});
