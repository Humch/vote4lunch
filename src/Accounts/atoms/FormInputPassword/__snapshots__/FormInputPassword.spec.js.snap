// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputPassword Snapshot capturing Snapshot of FormInputPassword with repeater prop false 1`] = `
<input
  className="form-control "
  id="password"
  name="password"
  onChange={[Function]}
  required={true}
  type="password"
  value=""
/>
`;

exports[`FormInputPassword Snapshot capturing Snapshot of FormInputPassword with repeater prop true 1`] = `
<input
  className="form-control "
  id="passwordRepeater"
  name="passwordRepeater"
  onChange={[Function]}
  required={true}
  type="password"
  value=""
/>
`;
